tests/lib/nested.sh

```diff
@@ -797,13 +798,25 @@ EOF
             else 
                 UBUNTU_IMAGE_CHANNEL_ARG=""
             fi
-            # ubuntu-image creates sparse image files
-            # shellcheck disable=SC2086
-            "$UBUNTU_IMAGE" snap --image-size 10G "$NESTED_MODEL" \
-                $UBUNTU_IMAGE_CHANNEL_ARG \
-                --output-dir "$NESTED_IMAGES_DIR" \
-                $EXTRA_FUNDAMENTAL \
-                $EXTRA_SNAPS
+
+            if [ -n "$NESTED_UBUNTU_IMAGE_PRESEED_KEY" ]; then
+                # ubuntu-image creates sparse image files
+                # shellcheck disable=SC2086
+                "$UBUNTU_IMAGE" snap --image-size 10G --preseed --preseed-sign-key=\""$NESTED_UBUNTU_IMAGE_PRESEED_KEY"\" "$NESTED_MODEL" \
+                    $UBUNTU_IMAGE_CHANNEL_ARG \
+                    --output-dir "$NESTED_IMAGES_DIR" \
+                    $EXTRA_FUNDAMENTAL \
+                    $EXTRA_SNAPS
+            else
+                # ubuntu-image creates sparse image files
+                # shellcheck disable=SC2086
+                "$UBUNTU_IMAGE" snap --image-size 10G "$NESTED_MODEL" \
+                    $UBUNTU_IMAGE_CHANNEL_ARG \
+                    --output-dir "$NESTED_IMAGES_DIR" \
+                    $EXTRA_FUNDAMENTAL \
+                    $EXTRA_SNAPS
+            fi
+
```

Comment:
would this work or will quoting break it?
```suggestion
            if [ -n "$NESTED_UBUNTU_IMAGE_PRESEED_KEY" ]; then
                UBUNTU_IMAGE_PRESEED_ARG=--preseed --preseed-sign-key=\""$NESTED_UBUNTU_IMAGE_PRESEED_KEY"\" 
            fi
            # ubuntu-image creates sparse image files
            # shellcheck disable=SC2086
            "$UBUNTU_IMAGE" snap --image-size 10G "$NESTED_MODEL" \
                $UBUNTU_IMAGE_CHANNEL_ARG \
                $UBUNTU_IMAGE_PRESEED_ARG \
                --output-dir "$NESTED_IMAGES_DIR" \
                $EXTRA_FUNDAMENTAL \
                $EXTRA_SNAPS
```

Reply:
Unfortunately it fails with:
```
/home/gopath/src/github.com/snapcore/snapd/tests/lib/nested.sh: line 803: --preseed-sign-key=" (test)": command not found
```

Trying to quote this assignment makes it fail when calling actual ubuntu-image.

Reply:
This should work:
```bash
declare -a UBUNTU_IMAGE_PRESEED_ARGS
if [ -n "$NESTED_UBUNTU_IMAGE_PRESEED_KEY" ]; then
    UBUNTU_IMAGE_PRESEED_ARGS+=(--preseed  --preseed-sign-key=\""$NESTED_UBUNTU_IMAGE_PRESEED_KEY"\")
fi
"$UBUNTU_IMAGE" snap --image-size 10G "$NESTED_MODEL" \
                $UBUNTU_IMAGE_CHANNEL_ARG \
                "${UBUNTU_IMAGE_PRESEED_ARGS[@]}" \
                --output-dir "$NESTED_IMAGES_DIR" \
                $EXTRA_FUNDAMENTAL \
                $EXTRA_SNAPS
```

Reply:
Interesting, will try, thanks!

Reply:
Yeah, it worked! Thanks!